tests/unittests/benchmark/task/profet/test_profet_task.py

```diff
@@ -0,0 +1,455 @@
+        # version.
+
+        model_config = self.Task.ModelConfig()
+
+        real_input_dir: Path = REAL_PROFET_DATA_DIR
+        real_x, real_y, real_c = model_config.load_data(real_input_dir)
+
+        fake_input_dir: Path = tmp_path_factory.mktemp("profet_data")
+        fake_x, fake_y, fake_c = model_config.load_data(fake_input_dir)
+        assert real_x.shape == fake_x.shape
+        assert real_y.shape == fake_y.shape
+        assert real_c.shape == fake_c.shape
+
+        assert real_x.dtype == fake_x.dtype
+        assert real_y.dtype == fake_y.dtype
+        assert real_c.dtype == fake_c.dtype
+
+        assert 0 <= real_x.min() and 0 <= fake_x.min()
+        assert real_x.max() <= 1 and fake_x.max() <= 1
+
+        min_y, max_y = y_min[model_config.benchmark], y_max[model_config.benchmark]
+        assert min_y <= real_y.min() and min_y <= fake_y.min()
+        assert real_y.max() <= max_y and fake_y.max() <= max_y
+
+        min_c, max_c = c_min[model_config.benchmark], c_max[model_config.benchmark]
+        assert min_c <= real_c.min() and min_c <= fake_c.min()
+        assert real_c.max() <= max_c and fake_c.max() <= max_c
+
+    @pytest.mark.timeout(30)
+    @pytest.mark.parametrize("device_str", _devices)
+    def test_attributes(
+        self,
+        profet_train_config: MetaModelConfig,
+        profet_input_dir: Path,
+        device_str: str,
+        tmp_path_factory,
+    ):
+        """Simple test: Check that a task can be created and that its attributes are used and set
+        correctly.
+        """
+        max_trials = 123
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
+        device = torch.device(device_str)
+        task = self.Task(
+            max_trials=max_trials,
+            model_config=profet_train_config,
+            input_dir=profet_input_dir,
+            checkpoint_dir=checkpoint_dir,
+            device=device,
+        )
+        assert task.max_trials == max_trials
+        assert task.input_dir == profet_input_dir
+        assert task.checkpoint_dir == checkpoint_dir
+        assert isinstance(task.device, torch.device)
+        assert task.device == device
+        assert task.device.type == device_str
+        # NOTE: The class attributes might differ per task, but that's ok.
+        assert isinstance(task.model_config, self.Task.ModelConfig)
+        assert asdict(task.model_config) == asdict(profet_train_config)
+
+    @pytest.mark.timeout(30)
+    def test_instantiating_task(
+        self,
+        profet_train_config: MetaModelConfig,
+        profet_input_dir: Path,
+        tmp_path_factory: TempPathFactory,
+        monkeypatch: MonkeyPatch,
+    ):
+        """Tests that when instantiating multiple tasks with the same arguments, the
+        meta-model is trained only once.
+        """
+
+        checkpoint_dir: Path = tmp_path_factory.mktemp("checkpoints")
```

Comment:
Same question here for checkpoint dir.

Reply:
Need a fresh, empty directory. checkpoint_dir is session-scoped.

Reply:
Fixed in 61cf56505616cdee6da4874b04291b495467ad93